node/pkg/processor/processor.go

```diff
@@ -252,9 +326,12 @@ func (p *Processor) Run(ctx context.Context) error {
 		case m := <-p.signedInC:
 			p.handleInboundSignedVAAWithQuorum(ctx, m)
 		case <-cleanup.C:
-			p.handleCleanup(ctx)
+			if workerId == 1 {
+				cleanup.Reset(CleanupInterval)
+				p.handleCleanup(ctx)
+			}
 		case <-govTimer.C:
-			if p.governor != nil {
+			if p.governor != nil && workerId == 1 {
```

Comment:
Instead of always checking workerId == 1, how about splitting it up into a RunBasic and a RunConcurrent, as in https://github.com/wormhole-foundation/wormhole/blob/2972a68700ec2adfb0fe159c4513cc6fbf3f1206/node/pkg/processor/processor.go. 
I think the code would be cleaner that way, but you'll always have to run at least two workers: One for non-concurrent functions and one for potentially concurrent functions.

Reply:
I split it into a housekeeper and a worker.